### DISCUSSION TITLE
Where to send the report

### DISCUSSION THREAD

**Discussion by francescodiperna** [phax/peppol-reporting#10] [General]
Hi, thanks to Philip now I'am able to create the reports to send.
But in the guideline I did not find any information how to send the report and how to build the SBDH.
Did you know somethings?

Bye

**Comment by phax** [comment]
See https://docs.peppol.eu/edelivery/guidelines/reporting/2023-12-01%20Peppol%20Reporting%20-%20SP%20Operational%20Guideline%20v1.0.pdf chapter 6.3 :)

> The receiving Peppol ID of OpenPeppol is `9925:be0848934496`

**Reply by francescodiperna** [reply]
Thanks Philip, you are a precious resource :-)

**Reply by phax** [reply]
In the meantime, there is an updated link to v1.0.2: https://docs.peppol.eu/edelivery/guidelines/reporting/2024-01-15%20Peppol%20Reporting%20-%20SP%20Operational%20Guideline%20v1.0.2.pdf

**Reply by francescodiperna** [reply]
Hi Philip, I tried to send a report in a test environment, I got this error "Message failed : Identifier '9925:be0848934496' is not registered in SML."

**Reply by phax** [reply]
That is correct, because the participant ID is only registered in production. To test, please use the testbed or try sending to `9915:helger` (without any payload validation on receiver side)

**Reply by francescodiperna** [reply]
Thanks, I'm using you library to build the SBDH header, the document id and process id are the following:

```
String documentID = "urn:fdc:peppol:transaction-statistics-reporting:1.0::TransactionStatisticsReport##urn:fdc:peppol.eu:oo:trns:transaction-statistics-reporting:1::1.0";		
String processID = "urn:fdc:peppol.eu:edec:bis:reporting:1.0";
```

```
String documentID = "urn:fdc:peppol:end-user-reporting:1.0::EndUserReport##urn:fdc:peppol.eu:oo:trns:end-user-report:1::1.0";
String processID = "urn:fdc:peppol.eu:edec:bis:reporting:1.0";
	
```

```

final PeppolSBDHData aData = PeppolSBDHData.createUBL21
			(
					documento.getDocumentElement (),
		            PeppolIdentifierFactory.INSTANCE)
			            .setSenderWithDefaultScheme (sender)
			            .setReceiverWithDefaultScheme (receiver)
			            .setDocumentTypeWithBusdoxDocidQns (docType)
			            .setProcessWithDefaultScheme (procType)
			            .setCountryC1(countryCode);
		    									
		    // Create the SBDH document
		    final StandardBusinessDocument aSBD = new PeppolSBDHDocumentWriter().createStandardBusinessDocument (aData);
		    result = new SBDMarshaller().getAsString (aSBD);
		    

```

**Reply by phax** [reply]
That seems to be incorrect, because `createUBL21` is only for UBL 2.1 :)
Why are you creating the SBDH manually? When using `Phase4PeppolSender.builder ()` it will create the SBDH fully for you - you just need to provide document type ID etc. Via the `sbdDocumentConsumer(...)` callback you can even get access to the created SBDH document if you want it.

**Reply by francescodiperna** [reply]
I'm not using phase4 for sending, we use Oxalis As4.
For the report I do not have to use UBL 2.1?

Thanks

**Reply by phax** [reply]
Ah okay. No, UBL 2.1 is for BIS 3 or so. See https://docs.peppol.eu/edelivery/envelope/Peppol-EDN-Business-Message-Envelope-2.0.1-2023-08-17.pdf for how to fill out the fields.
Here is an example SBDH for TSR - adopt accordingly for EUSR:
```xml
  <sh:StandardBusinessDocumentHeader>
    <sh:HeaderVersion>1.0</sh:HeaderVersion>
    <sh:Sender>
      <sh:Identifier Authority="iso6523-actorid-upis">9915:phase4-test-sender</sh:Identifier>
    </sh:Sender>
    <sh:Receiver>
      <sh:Identifier Authority="iso6523-actorid-upis">9915:helger</sh:Identifier>
    </sh:Receiver>
    <sh:DocumentIdentification>
      <sh:Standard>urn:fdc:peppol:transaction-statistics-report:1.0</sh:Standard>
      <sh:TypeVersion>1.0</sh:TypeVersion>
      <sh:InstanceIdentifier>65ef091f-c8d2-42c3-962f-92c83898e426</sh:InstanceIdentifier>
      <sh:Type>TransactionStatisticsReport</sh:Type>
      <sh:CreationDateAndTime>2024-01-18T10:45:22.2+01:00</sh:CreationDateAndTime>
    </sh:DocumentIdentification>
    <sh:BusinessScope>
      <sh:Scope>
        <sh:Type>DOCUMENTID</sh:Type>
        <sh:InstanceIdentifier>urn:fdc:peppol:transaction-statistics-report:1.0::TransactionStatisticsReport##urn:fdc:peppol.eu:edec:trns:transaction-statistics-reporting:1.0::1.0</sh:InstanceIdentifier>
        <sh:Identifier>busdox-docid-qns</sh:Identifier>
      </sh:Scope>
      <sh:Scope>
        <sh:Type>PROCESSID</sh:Type>
        <sh:InstanceIdentifier>urn:fdc:peppol.eu:edec:bis:reporting:1.0</sh:InstanceIdentifier>
        <sh:Identifier>cenbii-procid-ubl</sh:Identifier>
      </sh:Scope>
      <sh:Scope>
        <sh:Type>COUNTRY_C1</sh:Type>
        <sh:InstanceIdentifier>AT</sh:InstanceIdentifier>
      </sh:Scope>
    </sh:BusinessScope>
  </sh:StandardBusinessDocumentHeader>
```

**Comment by iPriyanshuRanjan** [comment]
Hi @phax 
Is there any simulator available that can simulate the Peppol reporting service other then Peppol Testbed, allowing me to test the component ?

**Reply by phax** [reply]
Yes, you can send a Peppol Report to 9915:helger - but it will not check for validity on reception. You can only test the transmission part of it. hth
